### ISSUE TITLE
[a11y]: Screen Reader announcing as Search table with excessive information.

### ISSUE THREAD

**Issue by anamhussain94** [carbon-design-system/carbon#21176]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

v11.83.0

### Automated testing tool and ruleset

WCAG Mapping: 2.4.6

### Assistive technology

MAC OS Voiceover

### Description

Screen Reader announcing as Search table with excessive information.

When user navigates to search field , the screenreader reads "search" multiple time. So basically it reads aria-label, placeholder, type="search" .
Need provision to add a concise name for the search box that accurately describes its purpose.

### WCAG 2.1 Violation

WCAG Mapping: 2.4.6

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-search--default&addons=1&args=labelText:aaaaa;type:&panelRight=0&selectedKind=Accordion&selectedStory=Default&stories=1

### Steps to reproduce

1. Navigate to carbon/design [https://react.carbondesignsystem.com/?path=/story/components-search--default&addons=1&args=type:&panelRight=0&selectedKind=Accordion&selectedStory=Default&stories=1](Search)
2. Use a tool of your choice to navigate to the field and observe what screenreader read.

<img width="1728" height="1117" alt="Image" src="https://github.com/user-attachments/assets/c744f2f5-6b28-4dfe-b8d2-3005610c0ce0" />

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by heloiselui** [comment]
This issue was raised for MacOS, but we validated in Windows too. Issue still exists. Excessive information announced by screen reader.
